### PR TITLE
MVP First Version

### PR DESCRIPTION
**Pull request by kegor**
Replicated solution from https://github.com/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pull/1.

**TODO:**
- [ ] Update README.

### CODE REVIEW

File: src/client/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+## Adding capabilities to your workspace
+
+Nx supports many plugins which add capabilities for developing different types of applications and different tools.
+
+These capabilities include generating applications, libraries, .etc as well as the devtools to test, and build projects as well.
+
+Below are some plugins which you can add to your workspace:
+
+- [Angular](https://angular.io)
+  - `ng add @nrwl/angular`
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 22
           

Complexity increasing per file
==============================
- src/Products/Common/Util/Comparator/CaseInsensitiveComparer.cs  1
- src/Products/Common/Entity/Web/FileDescriptionEntity.cs  1
- src/Global.asax.cs  1
- src/Products/Metadata/Entity/FilePropertyName.cs  1
- src/Products/Common/Util/Comparator/FileDateComparator.cs  2
- src/Products/Common/Config/CommonConfiguration.cs  1
- src/client/apps/metadata-e2e/src/plugins/index.js  1
- src/App_Start/RouteConfig.cs  1
- src/Products/Common/Entity/Web/UploadedDocumentEntity.cs  1
- src/Startup.cs  1
- src/Products/Common/Entity/Web/LoadDocumentEntity.cs  1
- src/Products/Common/Util/Parser/YamlParser.cs  2
- src/Products/Common/Entity/Web/FilePropertyEntity.cs  1
- GroupDocs.Metadata.WebForms.Test/MetadataTest.cs  1
- src/Products/Common/Config/ApplicationConfiguration.cs  5
- src/Products/Common/Entity/Web/PostedDataEntity.cs  1
- src/Products/Common/Entity/Web/ExceptionEntity.cs  1
- src/Products/Common/Util/Comparator/FileNameComparator.cs  2
- src/Products/Common/Config/ServerConfiguration.cs  1
- src/Products/Common/Resources/Resources.cs  5
- src/Products/Common/Config/GlobalConfiguration.cs  1
- src/client/apps/metadata-e2e/src/support/app.po.ts  1
- src/App_Start/WebApiConfig.cs  1
- src/Products/Metadata/Controllers/MetadataApiController.cs  18
- src/Products/Common/Config/ConfigurationValuesGetter.cs  8
- src/Products/Metadata/Config/MetadataConfiguration.cs  3
- src/Metadata.aspx.cs  1
- src/Products/Common/Util/Directory/DirectoryUtils.cs  4
- src/Products/Common/Entity/Web/PageDescriptionEntity.cs  1
         

Clones added
============
- src/Products/Metadata/Controllers/MetadataApiController.cs  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156&bid=16781837)

File: src/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿namespace GroupDocs.Metadata.WebForms
+{
+    public partial class Startup {
```
Comment (bobkovalex) [inline src/Startup.cs]: Codacy Issue found: ['partial' is gratuitous in this context.](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/client/README.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+- [React](https://reactjs.org)
+  - `ng add @nrwl/react`
+- Web (no framework frontends)
+  - `ng add @nrwl/web`
+- [Nest](https://nestjs.com)
+  - `ng add @nrwl/nest`
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/client/README.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+- [Express](https://expressjs.com)
+  - `ng add @nrwl/express`
+- [Node](https://nodejs.org)
+  - `ng add @nrwl/node`
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/client/README.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+- [Angular](https://angular.io)
+  - `ng add @nrwl/angular`
+- [React](https://reactjs.org)
+  - `ng add @nrwl/react`
+- Web (no framework frontends)
+  - `ng add @nrwl/web`
+- [Nest](https://nestjs.com)
+  - `ng add @nrwl/nest`
+- [Express](https://expressjs.com)
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/client/README.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  - `ng add @nrwl/express`
+- [Node](https://nodejs.org)
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/client/README.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+Nx supports many plugins which add capabilities for developing different types of applications and different tools.
+
+These capabilities include generating applications, libraries, .etc as well as the devtools to test, and build projects as well.
+
+Below are some plugins which you can add to your workspace:
+
+- [Angular](https://angular.io)
+  - `ng add @nrwl/angular`
+- [React](https://reactjs.org)
+  - `ng add @nrwl/react`
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/client/README.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+- Web (no framework frontends)
+  - `ng add @nrwl/web`
+- [Nest](https://nestjs.com)
+  - `ng add @nrwl/nest`
+- [Express](https://expressjs.com)
+  - `ng add @nrwl/express`
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/client/README.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+Below are some plugins which you can add to your workspace:
+
+- [Angular](https://angular.io)
+  - `ng add @nrwl/angular`
+- [React](https://reactjs.org)
+  - `ng add @nrwl/react`
+- Web (no framework frontends)
+  - `ng add @nrwl/web`
+- [Nest](https://nestjs.com)
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/client/README.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+## Adding capabilities to your workspace
+
+Nx supports many plugins which add capabilities for developing different types of applications and different tools.
+
+These capabilities include generating applications, libraries, .etc as well as the devtools to test, and build projects as well.
+
+Below are some plugins which you can add to your workspace:
+
+- [Angular](https://angular.io)
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/Global.asax.cs [hunk 11]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Web;
+using System.Web.Routing;
+using System.Web.Http;
+
+namespace GroupDocs.Metadata.WebForms
+{
+    public class Global : HttpApplication
+    {
+        void Application_Start(object sender, EventArgs e)
```
Comment (bobkovalex) [inline src/Global.asax.cs]: Codacy Issue found: [Remove the unused private method 'Application_Start'.](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/client/README.md [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+  - `ng add @nrwl/angular`
+- [React](https://reactjs.org)
+  - `ng add @nrwl/react`
+- Web (no framework frontends)
+  - `ng add @nrwl/web`
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/client/README.md [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+
+These capabilities include generating applications, libraries, .etc as well as the devtools to test, and build projects as well.
+
+Below are some plugins which you can add to your workspace:
+
+- [Angular](https://angular.io)
+  - `ng add @nrwl/angular`
+- [React](https://reactjs.org)
+  - `ng add @nrwl/react`
+- Web (no framework frontends)
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/client/README.md (hunk 1 above)
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/Metadata.aspx.cs [hunk 14]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+
+namespace GroupDocs.Metadata.WebForms
+{
+    public partial class Metadata : System.Web.UI.Page
```
Comment (bobkovalex) [inline src/Metadata.aspx.cs]: Codacy Issue found: ['partial' is gratuitous in this context.](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/client/README.md (hunk 8 above)
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/client/README.md (hunk 12 above)
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/client/README.md (hunk 3 above)
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/client/README.md [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+
+Nx supports many plugins which add capabilities for developing different types of applications and different tools.
+
+These capabilities include generating applications, libraries, .etc as well as the devtools to test, and build projects as well.
+
+Below are some plugins which you can add to your workspace:
+
+- [Angular](https://angular.io)
+  - `ng add @nrwl/angular`
+- [React](https://reactjs.org)
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/client/README.md [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+
+Run `ng g @nrwl/angular:lib my-lib` to generate a library.
+
+> You can also use any of the plugins above to generate libraries as well.
+
+Libraries are sharable across libraries and applications. They can be imported from `@client/mylib`.
+
+## Development server
+
+Run `ng serve my-app` for a dev server. Navigate to http://localhost:4200/. The app will automatically reload if you change any of the source files.
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[no-literal-urls] Don’t use literal URLs without angle brackets](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/Metadata.aspx.cs [hunk 17]
```diff
@@ -0,0 +1,11 @@
...
+    {
+        protected void Page_Load(object sender, EventArgs e)
```
Comment (bobkovalex) [inline src/Metadata.aspx.cs]: Codacy Issue found: [Add a nested comment explaining why this method is empty, throw a 'NotSupportedException' or complete the implementation.](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

**Review by Pavel-Teplitsky — APPROVED** [review]

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- src/Products/Common/Util/Comparator/FileDateComparator.cs  2
- src/Products/Common/Entity/Web/FilePropertyEntity.cs  1
- GroupDocs.Metadata.WebForms.Test/MetadataTest.cs  1
- src/Products/Common/Config/GlobalConfiguration.cs  1
- src/App_Start/WebApiConfig.cs  1
- src/Metadata.aspx.cs  1
- src/Products/Common/Config/ApplicationConfiguration.cs  5
- src/Startup.cs  1
- src/client/apps/metadata-e2e/src/support/app.po.ts  1
- src/Products/Common/Resources/Resources.cs  5
- src/Products/Metadata/Config/MetadataConfiguration.cs  3
- src/Products/Common/Entity/Web/PageDescriptionEntity.cs  1
- src/Products/Common/Config/ServerConfiguration.cs  1
- src/Products/Common/Util/Directory/DirectoryUtils.cs  4
- src/Products/Metadata/Controllers/MetadataApiController.cs  18
- src/Products/Common/Util/Parser/YamlParser.cs  2
- src/Products/Common/Util/Comparator/FileNameComparator.cs  2
- src/Products/Metadata/Entity/FilePropertyName.cs  1
- src/Products/Common/Config/ConfigurationValuesGetter.cs  8
- src/Products/Common/Entity/Web/PostedDataEntity.cs  1
- src/App_Start/RouteConfig.cs  1
- src/Global.asax.cs  1
- src/Products/Common/Util/Comparator/CaseInsensitiveComparer.cs  1
- src/Products/Common/Entity/Web/LoadDocumentEntity.cs  1
- src/Products/Common/Entity/Web/UploadedDocumentEntity.cs  1
- src/Products/Common/Entity/Web/FileDescriptionEntity.cs  1
- src/client/apps/metadata-e2e/src/plugins/index.js  1
- src/Products/Common/Config/CommonConfiguration.cs  1
- src/Products/Common/Entity/Web/ExceptionEntity.cs  1
         

Clones added
============
- src/Products/Metadata/Controllers/MetadataApiController.cs  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156&bid=16781837)

File: src/Startup.cs (hunk 2 above)
Comment (bobkovalex) [inline src/Startup.cs]: Codacy Issue found: ['partial' is gratuitous in this context.](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/Metadata.aspx.cs (hunk 17 above)
Comment (bobkovalex) [inline src/Metadata.aspx.cs]: Codacy Issue found: [Add a nested comment explaining why this method is empty, throw a 'NotSupportedException' or complete the implementation.](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/Global.asax.cs (hunk 11 above)
Comment (bobkovalex) [inline src/Global.asax.cs]: Codacy Issue found: [Remove the unused private method 'Application_Start'.](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

File: src/Metadata.aspx.cs (hunk 14 above)
Comment (bobkovalex) [inline src/Metadata.aspx.cs]: Codacy Issue found: ['partial' is gratuitous in this context.](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Common/Util/Comparator/FileNameComparator.cs  2
- src/Products/Common/Config/GlobalConfiguration.cs  1
- src/Startup.cs  1
- src/Products/Common/Entity/Web/LoadDocumentEntity.cs  1
- src/Products/Common/Entity/Web/UploadedDocumentEntity.cs  1
- src/Products/Common/Config/ServerConfiguration.cs  1
- src/Products/Common/Resources/Resources.cs  5
- src/App_Start/WebApiConfig.cs  1
- src/Products/Common/Config/ApplicationConfiguration.cs  5
- src/Products/Common/Entity/Web/PageDescriptionEntity.cs  1
- src/Products/Metadata/Entity/FilePropertyName.cs  1
- src/client/apps/metadata-e2e/src/plugins/index.js  1
- src/Products/Common/Config/CommonConfiguration.cs  1
- src/App_Start/RouteConfig.cs  1
- GroupDocs.Metadata.WebForms.Test/MetadataTest.cs  1
- src/Products/Common/Util/Comparator/CaseInsensitiveComparer.cs  1
- src/Products/Metadata/Config/MetadataConfiguration.cs  3
- src/Metadata.aspx.cs  1
- src/Products/Common/Entity/Web/PostedDataEntity.cs  1
- src/Global.asax.cs  1
- src/Products/Common/Util/Comparator/FileDateComparator.cs  2
- src/Products/Common/Entity/Web/FileDescriptionEntity.cs  1
- src/Products/Common/Util/Parser/YamlParser.cs  2
- src/Products/Common/Entity/Web/ExceptionEntity.cs  1
- src/Products/Metadata/Controllers/MetadataApiController.cs  18
- src/Products/Common/Util/Directory/DirectoryUtils.cs  4
- src/Products/Common/Config/ConfigurationValuesGetter.cs  8
- src/Products/Common/Entity/Web/FilePropertyEntity.cs  1
- src/client/apps/metadata-e2e/src/support/app.po.ts  1
         

Clones added
============
- src/Products/Metadata/Controllers/MetadataApiController.cs  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-WebForms/pullRequest?prid=5102156&bid=16781837)

**Pull request merged**